build/compile_descriptor

```diff
@@ -1,62 +1,66 @@
+    case file:read_file_info(Target) of
+        {ok, #file_info{mtime = TargetMTime}} ->
+            lists:all(fun(File) ->
+                case file:read_file_info(File) of
+                    {ok, #file_info{mtime = MTime}} when MTime =< TargetMTime ->
+                        true;
+                    _ ->
+                        false
+                end
+                      end, CheckResources);
+        _ ->
+            false
+    end.
 
-# The || exit $? is needed, if sh is a bash, then with some
-# versions of bash with -e, execution does not terminte if a
-# subshell exits with non-zero exit code
+check_source(FileList, EchoString, Cmd) ->
+    case is_up_to_date(FileList) of
+        true ->
+            ok;
+        _ ->
+            io:format("~s~n",[EchoString]),
+            os:cmd(Cmd)
+    end.
 
-set -e
-
-if [ -d "$REBAR_DEPS_DIR/gpb" ]; then
-    # rebar3
-    EBIN="$REBAR_DEPS_DIR/gpb/ebin"
-    TEST="$REBAR_DEPS_DIR/gpb/test"
-    mkdir -p "$EBIN"
-else
-    # rebar2
-    EBIN="./ebin"
-    TEST="./test"
-fi
-
-is_up_to_date descr_src/gpb_descriptor.erl \
-          priv/proto3/google/protobuf/descriptor.proto || \
-    ( echo "Compiling descriptor.proto..."
-    erl +B -noinput -pa "$EBIN" \
-        -I priv/proto3/google/protobuf -o descr_src \
-    -modprefix gpb_ \
-        -s gpb_compile c descriptor.proto ) || exit $?
-is_up_to_date "$EBIN/gpb_descriptor.beam" descr_src/gpb_descriptor.erl || \
-    ( echo "Compiling gpb_descriptor.erl..."
-    erlc -Idescr_src -Iinclude -o "$EBIN" +debug_info \
-        descr_src/gpb_descriptor.erl ) || exit $?
-is_up_to_date "$EBIN/gpb_compile_descr.beam" \
-    descr_src/gpb_compile_descr.erl \
-    descr_src/gpb_descriptor.hrl || \
-    ( echo "Compiling gpb_compile_descr.erl..."
-    erlc -Idescr_src -Iinclude -o "$EBIN" +debug_info \
-    descr_src/gpb_compile_descr.erl ) || exit $?
-if [ -d "$TEST" ]; then
-    # eg: "rebar3 eunit" or "rebar2 eunit"
-    # but not "rebar3 compile"
-    # and not when compiling gpb as a dependency, since
-    # the test directory does not exist in the hex package.
-    is_up_to_date "$TEST/gpb_compile_descr_tests.beam" \
-        descr_src/gpb_compile_descr_tests.erl \
-        descr_src/gpb_descriptor.hrl || \
-        ( echo "Compiling gpb_compile_descr_tests.erl..."
-        erlc -Idescr_src -Iinclude -o "$TEST" +debug_info \
-        descr_src/gpb_compile_descr_tests.erl ) || exit $?
-fi
+main(_) ->
+    REBAR_DEPS_DIR = os:getenv("REBAR_DEPS_DIR"),
+    case filelib:is_dir(filename:join(REBAR_DEPS_DIR, "gpb")) of
+        true ->
+            Ebin = filename:join([REBAR_DEPS_DIR, "gpb", "ebin"]),
+            TEST = filename:join([REBAR_DEPS_DIR, "gpb", "test"]),
+            filelib:ensure_dir(Ebin);
+        _ ->
+            Ebin = "./ebin",
+            TEST = "./test"
+    end,
+    CheckList = [
+        {
+            ["descr_src/gpb_descriptor.erl", "priv/proto3/google/protobuf/descriptor.proto"]
```

Comment:
Do the forward slashes work on windows? Or does this need to be filename:join instead?